packages/react-accordion/Spec.md

```diff
@@ -0,0 +1,299 @@
+
+## Prior art
+
+As a part of the spec definitions in Fluent UI, a research effort has been made through [Open UI](https://open-ui.org/). The current research proposal is available as an open source contribution undergoing review ([research proposal](https://github.com/WICG/open-ui/pull/263))
+
+## Comparison of `@fluentui/react` and `@fluentui/react-northstar`
+
+- All mentions of v7 or v8 == `@fluentui/react` ([docsite](https://developer.microsoft.com/en-us/fluentui#/))
+- All mentions of v0 == `@fluentui/react-northstar` ([docsite](https://fluentsite.z22.web.core.windows.net/))
+
+There's no comparison to be done between the two libraries, since v8 hasn't implemented this component.
+
+## Issues
+
+- #16553
+- #16452
+- #12953
+- #12238
+
+## API
+
+The `Accordion` should implement a `children` based API as is the standard across all the surveyed alternatives as a part of Open UI research in [Prior Art](#prior-art). The component will leverage the use of `context` in the interaction and data flows of child components.
+
+Sample usages will be given in the following section of this document [Sample code](#sample-code)
+
+### Accordion
+
+The root level component serves context and common API between all children.
+
+| Prop name          | Type                    | Default Value | Details                                                      |
+| ------------------ | ----------------------- | ------------- | ------------------------------------------------------------ |
+| expand (?)         | boolean                 | false         | Allows multiple panels to be expanded                        |
+| collapse (?)       | boolean                 | false         | Allows multiple panels to be collapsed                       |
+| expandIconPosition | Enum                    | "end"         | Position of the icon to indicate expansion                   |
+| heading            | ElementType             | "h3"          | Heading element for the panel                                |
+| onToggle           | (open: boolean) => void |               | Dispatched when it's state changes between opened and closed |
+
+By default, the Accordion must have one, and **ONLY** one (1..1), opened panel all the time.
+If `collapse` is used, then the Accordion can have one or zero (0..1) panels opened.
+If `expand` is used, then the Accordion can have one or more (1..n) panels opened.
+If both `expand` and `collapse` are used, then the Accordion doesn't have restrictions (0..n).
+
+### AccordionPanel
+
+| Prop name   | Type           | Details                                                                                                    |
+| ----------- | -------------- | ---------------------------------------------------------------------------------------------------------- |
+| title       | ShorthandValue | Label for representing a section of content that also serves as control for showing and hiding the content |
```

Comment:
Would it make sense to try to remove reduce the number shorthand props by adopting appropriate child components e.g `AccordionPanelTitle` and `AccordionPanelBody` ?

Reply:
My problem with that is the order of those child components. What do you think? @ling1726

Reply:
Yeah it's a tough one. In this case my trust would be for clear documentation and common sense that a header should be positioned above the content, lots of `Card` components out there do it similarly